datasets/tj/companies/crawler.py

```diff
@@ -0,0 +1,210 @@
+    context: Context,
+    section: str,
+    params: Dict[str, str],
+    page_number: int,
+) -> int:
+    """
+    Crawls a single page of the Tadjikistan data portal and emits the data.
+    Args:
+        context: The context object for the current dataset.
+        section: section of the registry (currently there are three)
+        params: The form params to use for the request.
+        page_number: The page number to crawl.
+    Returns:
+        The total number of pages.
+    """
+    entity_type = "Company"
+    if section == "physical":
+        entity_type = "Person"
+
+    local_params = params.copy()
+    local_params["__CALLBACKPARAM"] = get_callback_pagination_params(page_number)
+
+    response = context.fetch_text(
+        url=context.data_url.format(section=section),
+        method="POST",
+        data=local_params,
+        cache_days=CACHE_DAYS,
+    )
+
+    _, raw_payload = response.split("(", 1)
+    raw_payload = raw_payload.strip("()")
+    matches = re.search(r"'result':'(.*)'", raw_payload, re.M | re.S)
+
+    if not matches:
+        return 0
+
+    dom = html.fromstring(matches.group(1))
+    pages = [
+        el.text
+        for el in dom.xpath(".//td[contains(@class, 'dxpPageNumber_Office2003Blue')]")
+    ]
+
+    try:
+        max_page = max(map(lambda x: int(x.strip("[]")), pages))
+    except ValueError:
+        context.log.warning(f"Cannot find max page among pages {pages}")
+
+    for tr in dom.xpath(
+        "descendant-or-self::*[@id = 'ASPxGridView1_DXMainTable']/"
+        "descendant-or-self::*/tr[@class and contains(concat(' ', "
+        "normalize-space(@class), ' '), ' dxgvDataRow_Office2003Blue')]"
+    ):
+        cells = tr.xpath("td")
+        if len(cells) != 5:
+            context.log.warning(
+                f"Cannot parse a row {values}, skipping the rest of the page"
+            )
+            break
+
+        values = [cell.text.strip() for cell in cells]
+        item = dict(zip(["ein", "inn", "name", "date_of_reg", "status"], values))
+
+        item["date_of_reg"] = h.parse_date(
+            item["date_of_reg"],
+            ["%d.%m.%Y"],
+        )[0]
+
+        entity = context.make(entity_type)
+        entity.id = context.make_id(f"TJ{entity_type}", item["ein"], item["inn"])
+        entity.add("name", item["name"], lang="tgk")
+        entity.add("incorporationDate", item["date_of_reg"])
+        entity.add("status", item["status"], lang="rus")
+        entity.add("country", "tj")
+
+        # I'm not sure if I'm right about mapping of this two fields, that's all I found
+        # https://andoz.tj/docs/postanovleniya-pravitelstvo/Resolution__№323_ru.pdf
+        # Раками Мушаххаси Андозсупоранда (РМА/INN) Идентификационный Номер Плательщика, Taxpayer Identification Number
+        # Раками Ягонаи Мушаххас (РЯМ/EIN) Уникальный уникальный номер, Unique Unique Number
+        entity.add("taxNumber", item["inn"])
+        entity.add("registrationNumber", item["ein"])
```

Comment:
Here's a translation
https://learn.microsoft.com/en-us/partner-center/reg-number-id#:~:text=Tax%20Identification%20number-,tajikistan,-INN%20%E2%80%93%20Tax%20Identification
INN – Tax Identification number
EIN – Employer Identification number

Reply:
I'll add that to comments section. Do you also want me to map that ein into the different field of the entity?